api/wb/views.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import generics, status
+from rest_framework.exceptions import NotFound
+
+from osf.models import AbstractNode
+from addons.osfstorage.models import OsfStorageFileNode, OsfStorageFolder
+from api.base.utils import get_object_or_error
+from api.base.parsers import HMACSignedParser
+from api.wb.serializers import (
+    WaterbutlerMetadataSerializer
+)
+
+class FileMetadataMixin(generics.CreateAPIView):
+    """
+    Mixin with common code for WB move/copy hooks
+    """
+    parser_classes = (HMACSignedParser,)
+    serializer_class = WaterbutlerMetadataSerializer
+    view_category = 'wb'
+    node_lookup_url_kwarg = 'node_id'
+
+    # Overrides CreateAPIView
+    def get_object(self):
+        return self.get_node(self.kwargs[self.node_lookup_url_kwarg])
+
+    def get_node(self, node_id):
+        node = get_object_or_error(
+            AbstractNode,
+            node_id,
+            self.request,
+            display_name='node'
+        )
+        if node.is_registration:
+            raise NotFound
+        return node
+
+    def perform_create(self, serializer, action):
```

Comment:
It's an antipattern to change a method's signature in a subclass if you can avoid it (because you're breaking an API contract).

Better to use a new method name. Maybe `perform_file_action`?